evcouplings/complex/protocol.py

```diff
@@ -111,27 +315,373 @@ def _modify_segments(seg_list, seg_prefix):
+        kwargs["first_annotation_file"],
+        kwargs["first_identities_file"],
+        kwargs["first_focus_sequence"],
+        kwargs["first_alignment_file"]
+    )
+
+    most_similar_in_species_2, id_to_header_2 = _load_monomer_info(
+        kwargs["second_annotation_file"],
+        kwargs["second_identities_file"],
+        kwargs["second_focus_sequence"],
+        kwargs["second_alignment_file"]
+    )
+
+    # merge the two dataframes to get all species found in 
+    # both alignments
+    species_intersection = most_similar_in_species_1.merge(
+        most_similar_in_species_2,
+        how="inner",  # takes the intersection
+        on="species",  # merges on species identifiers
+        suffixes=("_1", "_2")
+    )
+
+    raw_alignment_file = prefix + "_raw.fasta"
+
+    # write the raw concatenated alignment
+    target_seq_id, target_seq_index = write_concatenated_alignment(
+        species_intersection,
+        id_to_header_1, id_to_header_2,
+        kwargs["first_alignment_file"],
+        kwargs["second_alignment_file"],
+        kwargs["first_focus_sequence"],
+        kwargs["second_focus_sequence"],
+        raw_alignment_file
+    )
+
+    # filter the raw concatenated alignment
+    raw_ali = Alignment.from_file(open(raw_alignment_file))
+    aln_outcfg, _ = modify_alignment(
+        raw_ali,
+        target_seq_index,
+        target_seq_id,
+        kwargs["first_region_start"],
+        **kwargs
+    )
+
+    def _modify_segments(seg_list, seg_prefix):
+        # extract segments from list representation into objects
+        segs = [
+            Segment.from_list(s) for s in seg_list
+            ]
+        # update segment IDs
+        for i, s in enumerate(segs, start=1):
+            s.segment_id = "{}_{}".format(seg_prefix, i)
+
+        return segs
+
+    # merge segments - this allows to have more than one segment per
+    # "monomer" alignment
+    segments_1 = _modify_segments(kwargs["first_segments"], "A")
+    segments_2 = _modify_segments(kwargs["second_segments"], "B")
+    segments_complex = segments_1 + segments_2
+
+    # make sure we return all the necessary information:
+    # * alignment_file: final concatenated alignment that will go into plmc
+    # * focus_sequence: this is the identifier of the concatenated target
+    #   sequence which will be passed into plmc with -f
+    outcfg = aln_outcfg
+    outcfg["segments"] = [s.to_list() for s in segments_complex]
+    outcfg["focus_sequence"] = target_seq_id
+
+    # save basic statistics of the concatenation
+    outcfg["concatentation_statistics_file"] = prefix + "_concatenation_statistics.csv"
+    describe_concatenation(
+        kwargs["first_annotation_file"],
+        kwargs["second_annotation_file"],
+        kwargs["first_genome_location_file"],
+        kwargs["second_genome_location_file"],
+        outcfg["concatentation_statistics_file"]
+    )
 
     return outcfg
 
 
+def best_reciprocal_hit(**kwargs):
```

Comment:
can this protocol be merged with best_hit()? look very similar to me, would save a lot of code duplication, and best reciprocal hit filtering could be turned on by a configuration flag instead